### PR TITLE
Fix catalogue item type interfaces #936

### LINKED ISSUE #936: Fix catalogue item type interfaces

### PR DESCRIPTION
**Pull request by joshuadkitenge**
## Description

fix the api types so they exactly match the backend 

## Testing instructions

Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking

closes #936

### CODE REVIEW

**Comment by joshuadkitenge** [comment]
sqaush/merge

File: src/api/api.types.tsx [hunk 1]
```diff
@@ -165,3 +165,73 @@ export interface CatalogueCategory
...
+  > {
+  catalogue_category_id?: string | null;
+  manufacturer_id?: string | null;
+  name?: string | null;
+  cost_gbp?: number | null;
+  days_to_replace?: number | null;
+  is_obsolete?: boolean;
+}
+
+export interface CatalogueItem extends CreatedModifiedMixin {
```
[the remark this answers is not in the document]
Comment (joshuadkitenge) [reply src/api/api.types.tsx:219]: it adds to much lines of code. I would have to omit all of the optional ones then re-add it without the  `?` and there are 10 optional parameters

[the remark this answers is not in the document]
Comment (joelvdavies) [reply src/api/api.types.tsx:220]: Oh right, I thought it worked in this case without omit? E.g.
https://www.typescriptlang.org/play/?#code/JYOwLgpgTgZghgYwgAgCoQM5gIzIN4BQyyAbnADYCuEA-AFzJZSgDmyAPsiJeeQQL4ECoSLEQp0WAEzIIAD0ggAJhjSYc+IqQrUGTVhy48+ggggD2ILMkhYGksDIC8+MlQgMARAAkIvc56CFlbm5BAAdOTmLAAUtmAAlEA
works while
https://www.typescriptlang.org/play/?#code/JYOwLgpgTgZghgYwgAgCoQM5gIzIN4BQyyAbnADYCuEAXMllKAObIA+yIl55BAvgQVCRYiFOiwAmZBAAekEABMMaTDnxFSFagH46DZmw5ce-AggD2ILMkhY64sFIC8+MlVrIARAAkI3c578Flbm5BAAdOTmTAAUtmAAlEA
doesn't as you found before. So I thought this case would work, unless it was a different issue?

[the remark this answers is not in the document]
Comment (joshuadkitenge) [reply src/api/api.types.tsx:220]: It does however for consistency I still omit the parameters. Once all of the RHF dialogs are done I'm going to optimise the interfaces ad types at the end. #287

**Review by joelvdavies — APPROVED** [review]

**Pull request merged**